pkg/cmd/pulumi/new.go

```diff
@@ -142,6 +165,44 @@ func runNew(ctx context.Context, args newArgs) error {
 		}
 	}
 
+	if args.templateNameOrURL == "" {
+		// Try to read the current project
+		project, _, err := readProject()
+		if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+			return err
+		}
+
+		b, err := currentBackend(ctx, project, opts)
+		if err != nil {
+			return err
+		}
+
+		name, _, _, err := b.CurrentUser()
+		if err != nil {
+			return err
+		}
+		var aiOrTemplate string
+		if shouldPromptForAIOrTemplate(args, b) {
+			aiOrTemplate, err = chooseWithAIOrTemplate(opts)
+		} else {
+			aiOrTemplate = deriveAIOrTemplate(args)
+		}
+		if err != nil {
+			return err
+		}
+		if aiOrTemplate == "ai" {
+			if b.Name() != "pulumi.com" {
+				// Currently, we don't actually hit this - users are prompted to log in
+				// for pulumi new regardless, but if that behavior changes, we can catch it here.
+				return errors.New("please log in to Pulumi Cloud to use Pulumi AI")
```

Comment:
Note: We do hit this if you're logged-in to the filestate backend (or I guess if you're logged into some other non-Pulumi Cloud URL), e.g.:

```
[~/Documents/2023/12/11/newtesttest]$ pulumi login --local
Logged in to Justins-Pulumi-MacBook-Pro.local as justin (file://~)
[~/Documents/2023/12/11/newtesttest]$ pulumi new --ai "foo"
error: please log in to Pulumi Cloud to use Pulumi AI
```

So the comment isn't really accurate.

Reply:
Correct! I'd meant to remove this after testing (I hadn't considered filestate backends, so it makes sense to remove this comment).